tests/msc3391_test.go

```diff
@@ -0,0 +1,274 @@
+//go:build msc3391
+// +build msc3391
+
+// This file contains tests for deleting account data as
+// defined by MSC3391, which you can read here:
+// https://github.com/matrix-org/matrix-doc/pull/3391
+
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+
+	"github.com/tidwall/gjson"
+)
+
+const testAccountDataType = "org.example.test"
+
+var testAccountDataContent = map[string]interface{}{"test_data": 1}
+
+func TestRemovingAccountData(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	// Create a user to manipulate the account data of
+	aliceUserID := "@alice:hs1"
+	alice := deployment.Client(t, "hs1", aliceUserID)
+
+	// And create a room with that user where we can store some room account data
+	roomID := alice.CreateRoom(t, map[string]interface{}{})
+
+	for _, httpMethod := range []string{"PUT", "DELETE"} {
+		t.Run(fmt.Sprintf("Deleting a user's account data via %s works", httpMethod), func(t *testing.T) {
+			createAndDeleteAccountData(t, alice, httpMethod == "DELETE", nil)
+		})
+		t.Run("Deleting a user's room account data via ", func(t *testing.T) {
+			createAndDeleteAccountData(t, alice, httpMethod == "DELETE", &roomID)
+		})
+	}
```

Comment:
I don't think this loop is really buying us anything TBH.

```suggestion
	// Test deleting global account data.
	t.Run("Deleting a user's account data via DELETE works", func(t *testing.T) {
		createAndDeleteAccountData(t, alice, true, nil)
	})
	t.Run("Deleting a user's account data via PUT works", func(t *testing.T) {
		createAndDeleteAccountData(t, alice, false, nil)
	})

	// Test deleting room account data.
	t.Run("Deleting a user's room data via DELETE works", func(t *testing.T) {
		createAndDeleteAccountData(t, alice, true, &roomID)
	})
	t.Run("Deleting a user's room account data via PUT works", func(t *testing.T) {
		createAndDeleteAccountData(t, alice, false, &roomID)
	})
```